lib/queue.rb

```
@@ -1,31 +1,55 @@
+QUEUE_SIZE = 20
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(QUEUE_SIZE)
+    @front = @rear = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1
+      @front = 0
+      @rear = 1
+      @store[@front] = element
+    elsif @front == (@rear + 1) % QUEUE_SIZE
+      raise Error, "Queue is full"
+    else
+      new_rear = (@rear + 1) % QUEUE_SIZE
+      @store[@rear] = element 
+      @rear = new_rear 
+    end
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @front == @rear
```

You first need to check to see if front and rear are both -1, if so, then the queue is empty, and you should raise an error. 

Otherwise if front == rear then the queue is full.